### PR TITLE
Update codebase to ZAP 2.13

### PR DESCRIPTION
**Pull request by thc202**
Change all add-ons and `testutils` to use 2.13 (SNAPSHOT).
Update code accordingly (e.g. address deprecations).
Update tests to assert with the headers lower case.
Add `maxAlertsPerRule` to active scan job/plans, required as it's a new core active scan property.

### CODE REVIEW

File: addOns/network/src/test/java/org/zaproxy/addon/network/internal/client/HttpSenderImplUnitTest.java [hunk 1]
```diff
@@ -585,7 +585,7 @@ class Response {
         void shouldHaveDataReceived(SenderMethod method) throws Exception {
             // Given
             String responseHeader =
-                    "HTTP/1.1 500 Reason\r\nHeader1: HeaderValue\r\nX: y\r\nContent-Length: 13\r\n\r\n";
+                    "HTTP/1.1 500 Reason\r\nHeader1: HeaderValue\r\nX: y\r\ncontent-length: 13\r\n\r\n";
```
Comment (kingthorin) [inline addOns/network/src/test/java/org/zaproxy/addon/network/internal/client/HttpSenderImplUnitTest.java:588]: Lower the fake header too?

**Review by kingthorin — COMMENTED** [review]
Just that one thing on the synthetic header. Otherwise looks good.

File: addOns/network/src/test/java/org/zaproxy/addon/network/internal/client/HttpSenderImplUnitTest.java (hunk 1 above)
Comment (thc202) [reply addOns/network/src/test/java/org/zaproxy/addon/network/internal/client/HttpSenderImplUnitTest.java]: Done.

**Review by psiinon — APPROVED** [review]

**Review by ricekot — APPROVED** [review]
Thank you!

**Pull request merged**